akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala

```diff
@@ -0,0 +1,403 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.Behavior.StoppedBehavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, MutableStashBuffer }
+import akka.event.Logging
+import akka.persistence.Eventsourced.{ PendingHandlerInvocation, StashingHandlerInvocation }
+import akka.persistence.JournalProtocol._
+import akka.persistence._
+import akka.persistence.journal.Tagged
+import akka.persistence.typed.scaladsl.PersistentBehaviors._
+import akka.{ actor ⇒ a }
+
+import scala.annotation.tailrec
+import scala.collection.immutable
+
+abstract class EventsourcedRunning[Command, Event, State](
+  val context:                 ActorContext[Any],
+  val internalStash:           MutableStashBuffer[Any],
+  private var _lastSequenceNr: Long,
+  instanceId:                  Int, writerUuid: String // TODO WRAP THOSE INTO ONE 'identity'
+) extends MutableBehavior[Any]
+  with EventsourcedBehavior[Command, Event, State]
+  with EventsourcedStashManagement { same ⇒
+
+  import EventsourcedBehavior._
+  import akka.actor.typed.scaladsl.adapter._
+
+  protected val log = Logging(context.system.toUntyped, classOf[EventsourcedRunning[_, _, _]])
+
+  // -------- initialize --------
+
+  // -- invoked at the end of this class, once all class fields have been initialized --
+  private[persistence] def init(): Unit = {
+    log.info(s"INITIALIZED $persistenceId, unstashing...")
+    tryUnstash(context, this) // will continue looping through stash as possible (awaiting persists etc
+  }
```

Comment:
Why is this needed, how could it have anything stashed before the behavior constructor has completed?

Reply:
It almost always will have things stashed, since we start it and send a command to it.
Before it is able to handle that command, it could have been running minutes already (!) because it needs to do all of the persistence replay dance, which is now modeled in typed-style with multiple behaviours handling each of the phases: `EventsourcedRequestingRecoveryPermit` -> `EventsourcedRecoveringSnapshot` -> `EventsourcedRecoveringEvents` -> `EventsourcedRunning` it could have been receiving commands.

The stash is shared between them, and by the time you get to Running it could have many many commands stashed already

Reply:
This is the one place where it isn't really fine to run through the unstash as far as I can see.

Reply:
the returned behavior of `tryUnstash` is not used here, e.g. stopped

Reply:
Hah, got it... thanks for helping spot this. I knew it's going to be something trivial that I overlooked. Thanks!